extensions/amp-carousel/0.1/test-e2e/test-slidescroll-autoplay.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describes.endtoend(
+  'AMP carousel 0.1 slideChange on type="slide" with autoplay',
+  {
+    testUrl:
+      'http://localhost:8000/test/fixtures/e2e/amp-carousel/0.1/slidescroll-autoplay.html',
+    environments: ['single'],
+  },
+  async function(env) {
+    const ActionTrust = {LOW: '1', HIGH: '3'};
+    let controller;
+    /**
+     * Attach an event listener to page to capture a custom event.
+     * @param {string} type Event name.
+     * @return {!Promise}
+     */
+    function updateDomOn(type) {
+      return controller.driver.executeScript(type => {
+        document.addEventListener(type, e => {
+          const el = document.createElement('div');
+          el.classList.add('event');
+          el.textContent = e.data.actionTrust;
+          document.querySelector('#event-container').appendChild(el);
+        });
+      }, type);
+    }
```

Comment:
Try this
```suggestion
function updateDomOn(type) {
  return controller.driver.executeAsyncScript(type => {
    return new Promise((resolve) => {
      document.addEventListener(type, e => {
        resolve(e.data.actionTrust);
      });
    });
  }, type);
}
```

Reply:
Gave this a try but the tests timed out: `ScriptTimeoutError: script timeout` What did you have in mind here?

Reply:
I'm trying to see if there's a way to communicate the `actionTrust` to the tests without the need for us to communicate through a dom node. From the documentation it seemed like `executeAsyncScript` work for that. I'd give this another try to debug the `timeout` error:

```suggestion
function updateDomOn(type) {
  return controller.driver.executeAsyncScript("var callback = arguments[arguments.length - 1];document.addEventListener('slideChange', callback);");
}
```